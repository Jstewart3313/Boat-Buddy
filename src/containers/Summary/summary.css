.counters {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  width: 100%;
  margin: 20px 0px 0px 0px;
}

.one-count {
  border: 2px #FFFFFF solid;
  border-radius: 10px;
  padding: 0px 5px 0px 5px;
}

.how {
  margin: 75px 0px 0px 0px;
}

@media only screen and (max-width: 600px) {
  .counters {
    display: flex;
    flex-direction: column;
  }
}
