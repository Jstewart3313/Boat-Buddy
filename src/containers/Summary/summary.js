import React from 'react';
import { connect } from 'react-redux';

export const Summary = (props) => {

  return (
    <div className='summary'>
      <h2>Movies: {props.movieCount} </h2>
      <h2>Music: {props.musicCount} </h2>
      <h2>Shows: {props.tvShowCount} </h2>
      <p className='how'>HOW TO PLAY: Each month their will be a new list of hot titles for you to check out, encouraging anyone on the water to stay off of social media and expand your knowledge on new movies, music and tv-shows.</p>
      <p className='invite'>Invite your friends and win challenges by keeping your watch count sky high!</p>
    </div>
  )
}

const mapStateToProps = state => ({
  movieCount: state.movieCounter,
  tvShowCount: state.tvShowCounter,
  musicCount: state.musicCounter
})

export default connect(mapStateToProps, null)(Summary);