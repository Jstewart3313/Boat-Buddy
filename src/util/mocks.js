import { key } from "./key";

export const musicUrl = `https://cors-anywhere.herokuapp.com/https://tastedive.com/api/similar?k=${key}&info=1&type=music&limit=10&q=redhotchilipeppers`;

export const movieUrl = `https://cors-anywhere.herokuapp.com/https://tastedive.com/api/similar?k=${key}&info=1&type=movie&limit=10&q=deadpool`;

export const tvUrl = `https://cors-anywhere.herokuapp.com/https://tastedive.com/api/similar?k=${key}&info=1&type=tvshow&limit=10&q=discovery`;

export const mockResult = {
  Similar: {
    Info: [
      {
        Name: "Discovery",
        Type: "show",
        wTeaser:
          "Discovery was a television program geared towards children and teenagers, produced by ABC News. The program began in the fall of 1962 as a weekday series, and was later moved to Sunday mornings. The program was hosted by actor/announcer Frank Buxton and actress/vocalist Virginia Gibson. The show's original studio announcer was ABC staff announcer Bill Owen, who replaced Buxton as host in 1966, continuing through 1971.  The shows hosted by Buxton were mostly studio productions, done in black-and-white;  beginning with Owen, the shows were produced in color, and involved much travel to on-site locations.  The actual on-air title of the series was named according to each year it was produced, beginning with Discovery '62 and ending with Discovery '71 (syndicated reruns only had the title Discovery).\n",
        wUrl: "http://en.wikipedia.org/wiki/Discovery_(1960s_TV_series)",
        yUrl: "https://www.youtube-nocookie.com/embed/hC7IMj7WFyE",
        yID: "hC7IMj7WFyE"
      }
    ],
    Results: [
      {
        Name: "Genius",
        Type: "show",
        wTeaser:
          "\nGenius is an American anthology period drama television series developed by Noah Pink and  Kenneth Biller that premiered on April 25, 2017 on National Geographic.\nThe first season follows the life of Albert Einstein, from his early years, through his time as a patent clerk, to his later years as a physicist who developed the theory of relativity; the season is based on the 2007 book Einstein: His Life and Universe by Walter Isaacson. In April 2017, National Geographic renewed the series for a second season, which follows the life and artistry of Pablo Picasso and aired from April 24 to June 19, 2018. In April 2018, National Geographic renewed the series for a third season set to follow the life of writer Mary Shelley.\n",
        wUrl: "https://en.wikipedia.org/wiki/Genius_(U.S._TV_series)",
        yUrl: "https://www.youtube-nocookie.com/embed/SICLBlHizUY",
        yID: "SICLBlHizUY"
      }
    ]
  }
};

export const cleanData = [
  {
    Link: "https://www.youtube-nocookie.com/embed/SICLBlHizUY",
    id: "4e5b7406-3a5d-4ff3-8b6e-4105e452a36b",
    name: "Genius",
    type: "show"
  }
];

export const mockState = {
  movies: [
    {
      name: "who knows",
      title: "another title",
      Link: "www.Link.com"
    }
  ],
  music: [
    {
      name: "who knows",
      title: "another title",
      Link: "www.Link.com"
    }
  ],
  tvShow: [
    {
      name: "who knows",
      title: "another title",
      Link: "www.Link.com"
    }
  ],
  movieCounter: 0,
  musicCounter: 0,
  tvShowCounter: 0
};
