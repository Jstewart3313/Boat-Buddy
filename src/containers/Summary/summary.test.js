import React from "react";
import { shallow } from "enzyme";
import { Summary, mapStateToProps } from "./summary";
import * as mocks from "../../util/mocks";

describe("Summary", () => {
  it("should match the snapshot", () => {
    let wrapper = shallow(<Summary />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("mapStateToProps", () => {
  it("should return the expected object", () => {
    const expected = {
      movieCount: 0,
      tvShowCount: 0,
      musicCount: 0
    };
    expect(mapStateToProps(mocks.mockState)).toEqual(expected);
  });
});
