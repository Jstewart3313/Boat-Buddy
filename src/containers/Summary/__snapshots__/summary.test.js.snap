// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Summary should match the snapshot 1`] = `
<div
  className="summary"
>
  <h2>
    Movies: 
     
  </h2>
  <h2>
    Music: 
     
  </h2>
  <h2>
    Shows: 
     
  </h2>
  <p
    className="how"
  >
    HOW TO PLAY: Each month their will be a new list of hot titles for you to check out, encouraging anyone on the water to stay off of social media and expand your knowledge on new movies, music and tv-shows.
  </p>
  <p
    className="invite"
  >
    Invite your friends and win challenges by keeping your watch count sky high!
  </p>
</div>
`;
